File: client/src/modules/Battle/Logic/Weapons/Weapon.js
var Weapon = cc.Sprite.extend({

    ctor: function(player, pos, mapMgr, mapGUI, cellSize, res) {
        this._super(res);

        this.player = player
        this.currentPos = pos;
        this.mapMgr = mapMgr;
        this.mapGUI = mapGUI;
        this.cellSize = cellSize
        this.curDir = cc.p(1,0);

        this.setScale(0.6)
        this.setAnchorPoint(0, 0.1)
    },

    updateDir: function (direction, dt) {
        if (direction.x === 0 && direction.y === 0) return;
        let angle = cc.pToAngle(direction);
        this.curDir = direction;

        if (direction.x >= 0) {
            this.setRotation(- angle / Math.PI * 180);
        }

        if (direction.x < 0) {
            this.setRotation(-180 + angle / Math.PI * 180);
        }
    },

    updatePosLogic: function (currentPos) {
        this.currentPos = currentPos;
    },

    attack: function () {
    },

    getTimeUseWeapon: function () {
    },

    update: function (dt) {
    }
});

File: server/src/main/java/battle/model/weapons/Weapon.java
package battle.model.weapons;

import battle.BattleMgr;
import battle.MapLogic;
import battle.model.Character;
import battle.model.Coordinate;

public class Weapon {
    public Character player;
    public Coordinate currentPos;
    public MapLogic mapMgr;
    public BattleMgr mapGUI;
    public double cellSize;
    public Coordinate curDir;
    public int numUse;

    public Weapon(Character player, Coordinate pos, MapLogic mapMgr, BattleMgr mapGUI, double cellSize) {
        this.player = player;
        this.currentPos = pos;
        this.mapMgr = mapMgr;
        this.mapGUI = mapGUI;
        this.cellSize = cellSize;
        this.curDir = new Coordinate(1,0);
    }

    public void updateDir(Coordinate direction, double dt) {
        if (direction.getX() == 0 && direction.getY() == 0) return;
        this.curDir = direction;
    }

    public void updatePosLogic(Coordinate currentPos) {
        this.currentPos.setX(currentPos.getX());
        this.currentPos.setY(currentPos.getY());
    }

    public void attack() {}

    public int getTimeUseWeapon() {
        return -1;
    }

    public void update(double dt) {}
}
